File: Maps/K-thSmallestPrimeFraction/fraction.py
import heapq
from typing import List

class Solution:
    def kthSmallestPrimeFraction(self, arr: List[int], k: int) -> List[int]:
        fractions = []
        n = len(arr)
        for i in range(n-1):
            for j in range(i+1,n):
                fractions.append((-arr[i]/arr[j],(arr[i],arr[j])))

        totalFractions = len(fractions)
        pq = []
        for i in range(totalFractions):
            heapq.heappush(pq,fractions[i])
            while len(pq)>k: heapq.heappop(pq)
        ans = [pq[0][1][0],pq[0][1][1]]
        return ans

File: Maps/K-thSmallestPrimeFraction/fraction.cpp
#include <vector>
#include <queue>
using namespace std;

class Solution {
public:
    typedef pair<float,pair<int,int>> pfp;
    vector<int> kthSmallestPrimeFraction(vector<int>& arr, int k) {
        vector<pfp>fractions;
        int n = arr.size();
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                fractions.push_back({1.0 * arr[i]/arr[j],{arr[i],arr[j]}});
            }
        }
        int totalFractions = fractions.size();
        priority_queue<pfp>pq;
        for(int i=0;i<totalFractions;i++){
            pq.push(fractions[i]);
            while(pq.size()>k)pq.pop();
        }
        vector<int>ans = {pq.top().second.first,pq.top().second.second};
        return ans;
    }
};